### PR TITLE
Move `add_builtin` to SVM

### LINKED ISSUE #303: SVM entry point requires builtins registered in two different places
#### Problem

In order to use the SVM, we must register the builtins as it is now done in the `add_builtin` function in `bank.rs` ([link](https://github.com/anza-xyz/agave/blob/64412096829241e5c70d537403584dbc9e04ee93/runtime/src/bank.rs#L7089-L7098)), and then pass them to `load_and_execute_sanitized_transactions` as a vector of public keys.

#### Proposed Solution

Ideally, we would only register the builtins on the SVM once, obviating the vector of public keys passed to `load_and_execute_sanitized_transactions`.

### PR DESCRIPTION
**Pull request by LucasSte**
#### Problem

Presently, external SVM users must register builtins in two different places separately: the program cache and a vector passed to `load_and_process_sanitized_transactions`. These steps are burdensome and can lead to potential mistakes. In addition, the execution of programs should be self contained in the SVM, so the registration of builtins is semantically related to the SVM scope.

#### Summary of Changes

1. Moved `add_builtin` from `bank.rs` to the SVM folder.
2. `add_builtin_account` is now part of the trait `TransactionProcessingCallback`.
3. Moved `builtin_program_ids` from `struct Bank` to `struct TransactionBatchProcessor`.

Fixes #303

### CODE REVIEW

**Review by dmakarov — DISMISSED** [review]

**Comment by LucasSte** [comment]
@pgarg66 mentioned he wanted to have a look at these changes, so I'll wait for his review before merging.

File: svm/src/transaction_processor.rs
Comment (Lichtso) [inline svm/src/transaction_processor.rs]: `TODO` or `unimplemented!()` or `unreachable!()`?

Comment (LucasSte) [reply svm/src/transaction_processor.rs]: This function is never used in this test setting, so I left the `todo!()` there. Maybe a `unimplemented!()` sounds better, doesn't it?

File: svm/src/account_loader.rs
Comment (Lichtso) [inline svm/src/account_loader.rs]: There is still a `todo!()` left over here.

Comment (pgarg66) [reply svm/src/account_loader.rs]: @LucasSte if this is an optional trait method, maybe have a default implementation of it in the trait definition?

File: runtime/src/bank.rs
Comment (pgarg66) [inline runtime/src/bank.rs]: Except for a few tests, `must_replace` is always `false`. @Lichtso can this be removed, and assumed to be false always?

Comment (Lichtso) [reply runtime/src/bank.rs]: I think we can remove that flag.

Comment (LucasSte) [reply runtime/src/bank.rs]: > I think we can remove that flag.

I can make a separate PR for the flag removal. This way, the moved code won't shadow that change.

File: svm/src/account_loader.rs
Comment (LucasSte) [reply svm/src/account_loader.rs]: It is an optional method, but coming up with a default implementation is not straightforward. In order to add a builtin account, we need access to a generic accounts database, which developers will only provide when they create their own struct to replace Agave's `Bank`.

Comment (pgarg66) [reply svm/src/account_loader.rs]: The default implementation can be empty. So you don't need to implement for test with a todo in it. If the users of SVM do not need builtin they can ignore it.

**Review by Lichtso — APPROVED** [review]

**Review by pgarg66 — APPROVED** [review]

**Pull request merged**
